cmd/collectors/zapi/collector/zapi.go

```diff
@@ -269,28 +271,28 @@ func (me *Zapi) PollData() (*matrix.Matrix, error) {
 	fetch = func(instance *matrix.Instance, node *node.Node, path []string) {
 
 		newpath := append(path, node.GetNameS())
-		logger.Debug(me.Prefix, " > %s(%s)%s <%s%d%s> name=[%s%s%s%s] value=[%s%s%s]", color.Grey, newpath, color.End, color.Red, len(node.GetChildren()), color.End, color.Bold, color.Cyan, node.GetNameS(), color.End, color.Yellow, node.GetContentS(), color.End)
+		key := strings.Join(newpath, ".")
+		logger.Debug(me.Prefix, " > %s(%s)%s <%s%d%s> name=[%s%s%s%s] value=[%s%s%s]", color.Grey, key, color.End, color.Red, len(node.GetChildren()), color.End, color.Bold, color.Cyan, node.GetNameS(), color.End, color.Yellow, node.GetContentS(), color.End)
 
 		if value := node.GetContentS(); value != "" {
-			key := strings.Join(newpath, ".")
-			if metric := me.Matrix.GetMetric(key); metric != nil {
+			if label, has := me.instanceLabelPaths[key]; has {
+				instance.SetLabel(label, value)
+				logger.Trace(me.Prefix, "%slabel (%s) [%s] set value (%s)%s", color.Yellow, key, label, value, color.End)
+				count += 1
+			} else if metric := me.Matrix.GetMetric(key); metric != nil {
 				if err := metric.SetValueString(instance, value); err != nil {
-					//logger.Warn(me.Prefix, "%sskipped metric (%s) set value (%s): %v%s", color.Red, key, value, err, color.End)
+					logger.Error(me.Prefix, "%smetric (%s) set value (%s): %v%s", color.Red, key, value, err, color.End)
 					skipped += 1
 				} else {
-					//logger.Trace(me.Prefix, "%smetric (%s) set value (%s)%s", color.Green, key, value, color.End)
+					logger.Trace(me.Prefix, "%smetric (%s) set value (%s)%s", color.Green, key, value, color.End)
```

Comment:
this was commented out before - did you mean to leave it uncommented?

Reply:
> Can you add a unit test that demonstrates the problem and shows this fixes it? Especially since we plan on revisiting this

it's a bit hard, since the incoming data is from `pkg/api/ontapi/zapi`, I think we should add a method to simulate data, but this will require more time I am afraid

Reply:
what I could do is add a test for parsing the counter tree from the template, that would require less time and code changes

Reply:
The change you made was to the `PollData` method and how it uses `instanceLabelPaths`. Can you create a test that creates a simple matrix, creates a Zapi struct, calls the `PollData` method with the matrix, then checks the relevant data after the call to `PollData`?

This method makes a zapi call - if that was refactored to a separate method seems like the relevant change around `instanceLabelPaths` could be made.

Reply:
The problem is the ZAPI call, this will fail in the test since the test unit (and the `PollData` method) will not be able to connect to ZAPI

Reply:
That's why I was suggesting refactoring the zapi call into another method.